File: other/MT1051/week03/HW05/HW05_109601003.go
// Date: 2023/09/25
// HW: 05
// Author: 林群賀
// Student Number: 109601003

package main

import (
	"fmt"
	"math"
)

func calculateProbability(x int, lambdaValue float64) float64 {
	eToTheMinusLambda := math.Exp(-lambdaValue)
	lambdaPowerX := math.Pow(lambdaValue, float64(x))
	factorialX := 1.0

	for i := 1; i <= x; i++ {
		factorialX *= float64(i)
	}

	probability := (eToTheMinusLambda * lambdaPowerX) / factorialX
	return probability
}

func main() {
	lambdaValue := 0.26

	for x := 1; x <= 6; x++ {
		probability := calculateProbability(x, lambdaValue)
		fmt.Printf("P(x=%d, lambda=%.2f) = %.6f\n", x, lambdaValue, probability)
	}
}

File: mt/mt1051/week03/HW05/HW05_109601003.py
# -*- coding: utf-8 -*-
"""
Date: 2023/09/25
HW: 05
Author: 林群賀
Student Number: 109601003
"""

import math

def calculate_probability(x, lambda_value) -> float:
    e_to_the_minus_lambda = math.exp(-lambda_value)
    
    lambda_power_x = lambda_value ** x
    
    factorial_x = math.factorial(x)
    
    probability = (e_to_the_minus_lambda * lambda_power_x) / factorial_x
    
    return probability

def main() -> None:
    lambda_value = 0.26

    for x in range(1, 7):
        probability = calculate_probability(x, lambda_value)
        print(f'P(x={x}, lambda={lambda_value}) = {probability:.6f}')

if __name__ == '__main__':
    main()
